pkg/controller/storageprofile-controller.go

```diff
@@ -132,8 +133,22 @@ func (r *StorageProfileReconciler) reconcilePropertySets(sc *storagev1.StorageCl
 	return claimPropertySets
 }
 
+func (r *StorageProfileReconciler) reconcileCloneStrategy(sc *storagev1.StorageClass, clonestrategy *cdiv1.CDICloneStrategy) {
+	if sc.GetAnnotations()["cdi.kubevirt.io/clone-strategy"] != "" {
+		if clonestrategy == nil {
+			clonestrategy = nil // default to nil
+		}
+	}
+}
+
 func (r *StorageProfileReconciler) createEmptyStorageProfile(sc *storagev1.StorageClass) (*cdiv1.StorageProfile, error) {
```

Comment:
the `reconcileStorageProfile` is written as follows:
1. get the actual profile as storageProfile (or empty if not found) so you should not chnage the `createEmptyStorageProfile` unless you have some sensible and well documented defautls

2. gather facts from environment and apply to storageProfile (currently there is only one such method `reconcilePropertySets`) and you are adding a new one `reconcileCloneStrategy`. And here is the best place to apply all changes. It will work for a new obejct and for a changes during runtime, adding/removing of annotation and so on. 

3. Update the StorageProfile if anything was changed (that is why we keep a cloe of storage progile named `prevStorageProfile` - to detect if there are changes, and kubernets object needs to be updated)